Src/FluentAssertions/Equivalency/ConstraintEquivalencyStep.cs

```diff
@@ -0,0 +1,197 @@
+                }
+            }
+
+            bool matchingType = subject.GetType() == expectation.GetType();
+
+            AssertionScope.Current
+                .ForCondition(matchingType)
+                .FailWith($"Expected Constraint of type {expectation.GetType()}, but found {subject.GetType()}");
+
+            if (matchingType)
+            {
+                if ((subject is UniqueConstraint subjectUniqueConstraint)
+                 && (expectation is UniqueConstraint expectationUniqueConstraint))
+                {
+                    CompareConstraints(parent, context, subjectUniqueConstraint, expectationUniqueConstraint);
+                }
+                else if ((subject is ForeignKeyConstraint subjectForeignKeyConstraint)
+                      && (expectation is ForeignKeyConstraint expectationForeignKeyConstraint))
+                {
+                    CompareConstraints(parent, context, subjectForeignKeyConstraint, expectationForeignKeyConstraint);
+                }
+                else
+                {
+                    AssertionScope.Current
+                        .FailWith($"Don't know how to handle a Constraint of type {subject.GetType()}");
+                }
+            }
+
+            return true;
+        }
+
+        private static void CompareConstraints(IEquivalencyValidator parent, IEquivalencyValidationContext context, UniqueConstraint subject, UniqueConstraint expectation)
+        {
+            AssertionScope.Current
+                .ForCondition(subject.ConstraintName == expectation.ConstraintName)
+                .FailWith($"Expected a constraint named {expectation.ConstraintName}, but found {subject.ConstraintName}");
+
+            parent.AssertEqualityUsing(
+                CreateNestedContext(context, nameof(subject.ExtendedProperties), subject.ExtendedProperties, expectation.ExtendedProperties));
+
+            AssertionScope.Current
+                .ForCondition(subject.IsPrimaryKey == expectation.IsPrimaryKey)
+                .FailWith($"Expected a {(expectation.IsPrimaryKey ? "Primary Key" : "Foreign Key")} constraint, but found a {(subject.IsPrimaryKey ? "Primary Key" : "Foreign Key")} constraint");
+
+            CompareConstraintColumns(subject.Columns, expectation.Columns);
+        }
+
+        private static void CompareConstraints(IEquivalencyValidator parent, IEquivalencyValidationContext context, ForeignKeyConstraint subject, ForeignKeyConstraint expectation)
+        {
+            AssertionScope.Current
+                .ForCondition(subject.ConstraintName == expectation.ConstraintName)
+                .FailWith($"Expected a constraint named {expectation.ConstraintName}, but found {subject.ConstraintName}");
+
+            parent.AssertEqualityUsing(
+                CreateNestedContext(context, nameof(subject.ExtendedProperties), subject.ExtendedProperties, expectation.ExtendedProperties));
+
+            // TODO: how to exclude TableName from comparison?
```

Comment:
🤔 What are you going to do with this?

Reply:
Ah, indeed. I think I can probably use the same pattern here that I used in other classes for handling selected members.